kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/http/HttpRequestProcessor.java

```diff
@@ -0,0 +1,140 @@
+import lombok.AllArgsConstructor;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+
+public abstract class HttpRequestProcessor implements AutoCloseable {
+
+    protected static final ObjectMapper MAPPER = new ObjectMapper()
+            .configure(SerializationFeature.INDENT_OUTPUT, true);
+
+    public static FullHttpResponse buildStringResponse(String body, String contentType) {
+        FullHttpResponse httpResponse = new DefaultFullHttpResponse(HTTP_1_1, OK,
+                Unpooled.copiedBuffer(body, CharsetUtil.UTF_8));
+        httpResponse.headers().set(HttpHeaderNames.CONTENT_TYPE, contentType);
+        httpResponse.headers().set(HttpHeaderNames.CONTENT_LENGTH, body.length());
+        addCORSHeaders(httpResponse);
+        return httpResponse;
+    }
+
+    public static FullHttpResponse buildEmptyResponseNoContentResponse() {
+        FullHttpResponse httpResponse = new DefaultFullHttpResponse(HTTP_1_1, NO_CONTENT, Unpooled.EMPTY_BUFFER);
+        httpResponse.headers().set(HttpHeaderNames.ALLOW, "GET, POST, PUT, DELETE");
+        httpResponse.headers().set(HttpHeaderNames.CONTENT_LENGTH, 0);
+        addCORSHeaders(httpResponse);
+        return httpResponse;
+    }
+
+    public static FullHttpResponse buildErrorResponse(HttpResponseStatus error, String body, String contentType) {
+        FullHttpResponse httpResponse = new DefaultFullHttpResponse(HTTP_1_1, error,
+                Unpooled.copiedBuffer(body, CharsetUtil.UTF_8));
+        httpResponse.headers().set(HttpHeaderNames.CONTENT_TYPE, contentType);
+        httpResponse.headers().set(HttpHeaderNames.CONTENT_LENGTH, body.length());
+        addCORSHeaders(httpResponse);
+        return httpResponse;
+    }
+
+    public static FullHttpResponse buildJsonErrorResponse(Throwable throwable) {
+        Throwable err = throwable;
+        while (err instanceof CompletionException) {
+            err = err.getCause();
+        }
+        int httpStatusCode = err instanceof SchemaStorageException
+                ? ((SchemaStorageException) err).getHttpStatusCode()
+                : INTERNAL_SERVER_ERROR.code();
+        HttpResponseStatus error = HttpResponseStatus.valueOf(httpStatusCode);
+
+        FullHttpResponse httpResponse;
+        try {
+            String body = MAPPER.writeValueAsString(new ErrorModel(httpStatusCode, err.getMessage()));
+            httpResponse = new DefaultFullHttpResponse(HTTP_1_1, error,
+                    Unpooled.copiedBuffer(body, CharsetUtil.UTF_8));
+            httpResponse.headers().set(HttpHeaderNames.CONTENT_TYPE, "application/vnd.schemaregistry.v1+json");
+            httpResponse.headers().set(HttpHeaderNames.CONTENT_LENGTH, body.length());
+            addCORSHeaders(httpResponse);
+        } catch (JsonProcessingException impossible) {
+            String body = "Error " + err;
+            httpResponse = new DefaultFullHttpResponse(HTTP_1_1, error,
```

Comment:
This is a duplicate of line 82. There should be a better way to organize this.

If there's no equivalent of `Try` class of scala, how about

```
String body = "Error " + err;
try {
    body = MAPPER.writeValueAsString(new ErrorModel(httpStatusCode, err.getMessage()));
} catch (JsonProcessingException impossible) {
} finally {
  httpResponse = ...
}
```